src/ansys/dpf/core/plotter.py

```diff
@@ -214,11 +214,23 @@ def add_field(
         **kwargs,
     ):
         # Get the field name
-        name = field.name.split("_")[0]
-        unit = field.unit
-        kwargs.setdefault("stitle", f"{name} ({unit})")
-
-        kwargs = self._set_scalar_bar_title(kwargs)
+        if isinstance(field, dpf.core.Field):
+            name = field.name.split("_")[0]
+            categories = False
+            unit = field.unit
+            kwargs.setdefault("stitle", f"{name} ({unit})")
+            kwargs = self._set_scalar_bar_title(kwargs)
+        elif isinstance(field, dpf.core.PropertyField):
+            try:
+                name = field.name
+            except dpf_errors.DpfVersionNotSupported:
+                name = ""
+            categories = True
+            kwargs.setdefault("stitle", f"{name}")
+            kwargs = self._set_scalar_bar_title(kwargs)
+            kwargs["scalar_bar_args"]["n_labels"] = len(set(field.data))
+            kwargs["scalar_bar_args"]["fmt"] = "%.0f"
```

Comment:
I would prefer to directly use the `add_scalar_bar` directly, I think something like this should work in this context
```suggestion
            self._plotter.add_scalar_bar(
                title='Your title',
                n_labels=len(set(field.data))
                fmt='%.0f',
)
```